### PR TITLE
Backport tiff.dll fix to v2.2.x

### PR DESCRIPTION
**Pull request by djhoese**
As discussed in #244 this backports the 2.3.x fix for the libtiff packages change from `libtiff.dll` to `tiff.dll`. This *should* fix python 2.7 environments on Windows.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by djhoese** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_python2.7.yaml [hunk 1]
```diff
@@ -13,31 +13,31 @@ docker_image:
 expat:
 - '2.2'
 gdal:
-- '2.2'
+- '2.3'
 geos:
-- 3.6.2
+- 3.7.0
```
Comment (ocefpaf) [inline .ci_support/linux_python2.7.yaml]: We may need to force this and others to older version to get vc9 working.
(Probably geos and hdf5, maybe proj.4 but I haven't checked.)

**Comment by djhoese** [comment]
I think I need to force hdf5 to 1.10.3 so that netcdf4 and h5py can be used in this same environment on Windows. Thoughts @ocefpaf ?

File: .ci_support/linux_python2.7.yaml (hunk 1 above)
Comment (djhoese) [reply .ci_support/linux_python2.7.yaml]: Yeah, looks like in my VM environment geos should be 3.6.2. I think hdf5 has to be 1.10.3 to get the most recent libnetcdf.

File: .ci_support/osx_python3.7.yaml [hunk 2]
```diff
@@ -13,31 +13,31 @@ cxx_compiler:
 expat:
 - '2.2'
 gdal:
-- '2.2'
+- '2.3'
```
Comment (djhoese) [inline .ci_support/osx_python3.7.yaml]: @ocefpaf where/how is this getting set?

Comment (ocefpaf) [reply .ci_support/osx_python3.7.yaml]: These are the pins used in the build. Let me send a PR to conda-forge-pinning, where this comes from, adding an optional pinning for `vc9`, that means we may not build things for `vc9` on the new `gcc7`, but they can still work after a re-render. Hold on...

**Comment by ocefpaf** [comment]
Let's get this in and I'll fix the vc9 build later.

**Pull request merged**